libp2p/muxers/mplex/lpchannel.nim

```diff
@@ -263,3 +212,105 @@ method close*(s: LPChannel) {.async, gcsafe.} =
 
   s.closedLocal = true
   asyncCheck closeInternal()
+
+proc timeoutMonitor(s: LPChannel) {.async.} =
+  ## monitor the channel for innactivity
+  ##
+  ## if the timeout was hit, it means that
+  ## neither incoming nor outgoing activity
+  ## has been detected and the channel will
+  ## be reset
+  ##
+
+  try:
+    while true:
+      await sleepAsync(s.timeout)
```

Comment:
This future should be started to allow cancellation.

I did something similar for gossip 1.1 (wip)

https://github.com/status-im/nim-libp2p/blob/gossip-one-one/libp2p/switch.nim#L576

Reply:
This is a tricky one because there is no one place to await for the timer task to finish, but we can and should cancel the timer itself. The problem with awaiting, is that the shutdown/close flow is not a single step but at least two in most cases. I need to think a little bit more about this.